python/ray/_private/node.py

```diff
@@ -177,9 +178,34 @@ def __init__(
 
         # Register the temp dir.
         if head:
-            # date including microsecond
-            date_str = datetime.datetime.today().strftime("%Y-%m-%d_%H-%M-%S_%f")
-            self._session_name = f"session_{date_str}_{os.getpid()}"
+            maybe_key = None
+            if self._ray_params.external_addresses is not None:
```

Comment:
Make this part a function and make it clear this has to be used to obtain session name?